src/shogun/labels/BinaryLabelEncoder.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ *
+ */
+#ifndef _BINARYLABELENCODER__H__
+#define _BINARYLABELENCODER__H__
+
+#include <memory>
+#include <shogun/base/SGObject.h>
+#include <shogun/labels/BinaryLabels.h>
+#include <shogun/labels/DenseLabels.h>
+#include <shogun/labels/LabelEncoder.h>
+#include <shogun/lib/SGVector.h>
+#include <unordered_set>
+namespace shogun
+{
+
+	class BinaryLabelEncoder : public LabelEncoder
+	{
+	public:
+		BinaryLabelEncoder() = default;
+
+		~BinaryLabelEncoder() = default;
+
+		/** Fit label encoder
+		 *
+		 * @param Target values.
+		 * @return SGVector which contains unique labels.
+		 */
+		SGVector<float64_t> fit(const std::shared_ptr<Labels>& labs) override
+		{
+			const auto result_vector = labs->as<DenseLabels>()->get_labels();
+			auto result_labels = fit_impl(result_vector);
+			require(
+			    unique_labels.size() == 2,
+			    "Binary Labels should contain only two elements");
```

Comment:
nitpick: "Binary labels" (the idea is to not replicate class/variable/type names in user facing error messages).
Still would be good to print the unique labels in this error message

Reply:
but the amount of the unique labels may be exceed more than two, such as if `{0,1,2,3,4,5,6,7,8}` is passed to the method, should we print all values?

Reply:
yeah the idea is that the user gets a feedback what is the the problem with his input. i.e. you print out that we've detected `{0,1,2}` as unique labels in his data set... this should help one to figure out/fix the input...

Reply:
++ what viktor says here.
Also keep in mind that the unique number of labels will be relatively small (unless someone passes regression labels .... so maybe you could cap them at length 10 or so and make a "...", although I don't really think that is necessary)